js/exporter/image_creator.js

```diff
@@ -419,31 +419,46 @@ function drawElement(element, context, parentOptions, shared) {
 
     if(!isText) {
         applyFilter(context, options, shared);
-        fillElement(context, options, shared);
+        if(!isImage) {
+            promise = fillElement(context, options, shared);
+        }
         strokeElement(context, options);
     }
 
-    applyGradient(context, options, shared, element);
+    applyGradient(context, options, shared, element, 'linear');
+    applyGradient(context, options, shared, element, 'radial');
 
     context.restore();
 
     return promise;
 }
 
-function applyGradient(context, options, { gradients }, element) {
-    if(gradients.length === 0) {
+function applyGradient(context, options, { linearGradients, radialGradients }, element, type) {
+    const gradients = type === 'linear' ? linearGradients : radialGradients;
+    if(Object.keys(gradients).length === 0) {
         return;
     }
     const id = parseUrl(options.fill);
     if(id && gradients[id]) {
         const box = element.getBBox();
-        const gradient = context.createLinearGradient(box.x, 0, box.x + box.width, 0);
-
-        gradients[id].forEach(opt => {
+        const horizontalCenter = box.x + box.width / 2;
+        const verticalCenter = box.y + box.height / 2;
+        const maxRadius = Math.max(box.height / 2, box.width / 2);
+        const gradient = type === 'linear' ?
```

Comment:
Nevermind, I agree that it should be fixed by eslint. But i thought it would be better to start sync now.
I especially referred to airbnb to not rely only on my feelings. Here is one more [mention](https://github.com/airbnb/javascript#comparison--nested-ternaries) right in airbnb guide, code sample on mdn [conditionals](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/conditionals) page, code convention from [Douglas Crockford](https://www.crockford.com/), sample in [JavaScript Standart Style](https://standardjs.com/rules.html).